src/3d/materials/qgsphongtexturedmaterialsettings.cpp

```diff
@@ -104,79 +114,92 @@ Qt3DRender::QMaterial *QgsPhongTexturedMaterialSettings::toMaterial( QgsMaterial
     case QgsMaterialSettingsRenderingTechnique::TrianglesFromModel:
     case QgsMaterialSettingsRenderingTechnique::TrianglesDataDefined:
     {
-      bool fitsInCache = false;
-      const QImage textureSourceImage = QgsApplication::imageCache()->pathAsImage( mDiffuseTexturePath, QSize(), true, 1.0, fitsInCache );
-      ( void )fitsInCache;
+      Qt3DRender::QMaterial *material = nullptr;
+      QImage textureSourceImage = mDiffuseTextureImage.copy();
```

Comment:
why are we doing explicit copy of the QImage here, and in QgsPhongTexturedMaterialSettings::setDiffuseTexture() ?